pkg/apis/openstackproviderconfig/v1alpha1/types.go

```diff
@@ -105,6 +128,51 @@ type RootVolume struct {
 	Size       int    `json:"diskSize,omitempty"`
 }
 
+// OpenstackProviderSpecTemp is a workaround for unmarshalling clusterv1 providerSpec
+// Todo(egarcia): come up with a better unmarshalling scheme
+type OpenstackProviderSpecTemp struct {
```

Comment:
TB, I'm not sure I understand the need for this struct. Happy to chat about it when you're online.